opds.py

```diff
@@ -650,7 +633,7 @@ def facet_links(self, annotator, facets):
 
 
     def __init__(self, _db, title, url, works, annotator=None,
-                 messages_by_urn={}, precomposed_entries=[]):
+                 precomposed_entries=[]):
```

Comment:
Here's the core of the change. Instead of a separate map of message data structures that get rendered into OPDS entries, `precomposed_entries` can now contain a mixed list of <entry> tags and OPDSMessage objects.